internal/controllers/scheduling/op.go

```diff
@@ -0,0 +1,233 @@
+package scheduling
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"hash/fnv"
+	"reflect"
+	"time"
+
+	apiv1 "github.com/Azure/eno/api/v1"
+	"github.com/go-logr/logr"
+	"github.com/google/uuid"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type op struct {
+	Synthesizer *apiv1.Synthesizer
+	Composition *apiv1.Composition
+	Reason      opReason
+
+	Dispatched time.Time
+
+	id               uuid.UUID // set when patch is built
+	synthRolloutHash []byte    // memoized
```

Comment:
Any particular concurrency reason to memoize? Or is it just to avoid the hash computation?

Reply:
Just to avoid hashing. Because `SynthRolloutOrderHash` is called by `Less`, it will likely be called several times per op.